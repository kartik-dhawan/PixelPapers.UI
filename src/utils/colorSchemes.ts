export const GLOBAL_COLORS = {
  TEXT_PRIMARY_DARK: "#222",
  TEXT_PRIMARY_LIGHTER: "#666",
  TEXT_PRIMARY_LIGHT: "#d9d9d9",
  BACKGROUND_PRIMARY_LIGHT: "#fefefe",
  BACKGROUND_PRIMARY_DARK: "#111",
  COMMON_FOOTER_BACKGRUND_DARK: "#383b3e",
  BLOG_SUB_COMPONENT_BACKGROUND_LIGHTER: "#efefef",
}

export const PAGE_LEVEL_COLORS = {
  PAGE_404: {
    BACKGROUND_PRIMARY_LIGHT: "#226bb152",
    TEXT_PRIMARY_LIGHT: "#444",
    TEXT_PRIMARY_DARK: "#1c354c",
  },
  TRAVEL: {
    BACKGROUND_PRIMARY_LIGHT: "#fdeded",
    BACKGROUND_PRIMARY_DARK: "#864040",
    TEXT_PRIMARY_LIGHTER: "#444",
  },
}
