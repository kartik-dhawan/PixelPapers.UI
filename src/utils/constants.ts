export const CURATED_BY_LINK_DEFAULT_TEXT = "kartik-dhawan"

export const AVERAGE_TIME_TO_READ_A_BLOG /**In minutes */ = 7
