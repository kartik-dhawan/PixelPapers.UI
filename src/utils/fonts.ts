import { DM_Sans, Anton, Libre_Barcode_39 } from "next/font/google"

export const dmSans = DM_Sans({
  weight: ["200", "300", "400", "500", "600", "700"],
  subsets: ["latin"],
  display: "swap",
  preload: true,
})

export const barcode = Libre_Barcode_39({
  weight: ["400"],
  subsets: ["latin"],
  display: "swap",
})

export const anton = Anton({
  weight: ["400"],
  subsets: ["latin"],
  display: "swap",
})
