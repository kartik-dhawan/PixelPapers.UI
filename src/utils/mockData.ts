export const TRAVEL_PAGE_THUMBNAIL_CARD_DATA = {
  blogTitle: "Top 10 Things to do on your next trip to Jibhi/Jalori Pass.",
  blogMetaImage: {
    metadata: {
      tags: [],
    },
    sys: {
      space: {
        sys: {
          type: "Link",
          linkType: "Space",
          id: "1v98waf66sfj",
        },
      },
      id: "6TVtXnjJIZ812Bxm6VBqz5",
      type: "Asset",
      createdAt: "2023-11-21T17:39:08.757Z",
      updatedAt: "2023-11-21T17:39:17.165Z",
      environment: {
        sys: {
          id: "master",
          type: "Link",
          linkType: "Environment",
        },
      },
      revision: 2,
      locale: "en-US",
    },
    fields: {
      title: "Jibhi_Jalori",
      description:
        "A picture of sunset from Raghupur fort close to jalori pass",
      file: {
        url: "//images.ctfassets.net/1v98waf66sfj/6TVtXnjJIZ812Bxm6VBqz5/5b89ce3dced278570a2a663453d69a65/IMG_3749.webp",
        details: {
          size: 71536,
          image: {
            width: 3024,
            height: 4032,
          },
        },
        fileName: "IMG_3749.webp",
        contentType: "image/webp",
      },
    },
  },
  blogMetaTitle: "Jibhi - Jalori",
  blogTravelLocation: "Himachal Pradesh",
  blogType: "Travel Itinerary",
}
