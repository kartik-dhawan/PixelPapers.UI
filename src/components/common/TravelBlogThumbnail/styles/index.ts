import { GLOBAL_COLORS } from "@/utils/colorSchemes"

export const styles = {
  travelThumbnailsSectionWrapper: {
    width: "70%",
    minWidth: {
      xs: "100vw",
      md: "900px",
    },
  },
  travelThumbnailsImageList: {
    display: "flex !important",
    width: "100%",
    flexDirection: {
      xs: "column",
      md: "row",
    },
    padding: {
      xs: "0rem 1rem",
      sm: "0rem 3rem",
      md: "0rem 1rem",
    },
    boxSizing: "border-box",
  },
  travelThumbnailsCardListItem: {
    minWidth: "250px",
    transition: "1600ms all ease",
    "&:hover": {
      flexGrow: 7,
    },
  },
  travelThumbnailsImageWrapper: {
    position: "relative",
    height: {
      xs: "300px",
      sm: "425px",
      md: "500px",
    },
    width: "100%",
    "& img": {
      objectFit: "cover",
    },
  },
  travelThumbnailsCardTextWrapper: {
    color: GLOBAL_COLORS.TEXT_PRIMARY_LIGHT,
    background: "linear-gradient(180deg, #00000000 0%, #000 75%)",
  },
  travelThumbnailsBlogType: {
    fontSize: {
      xs: "14px",
      xl: "16px",
    },
    textTransform: "uppercase",
    fontWeight: 200,
    letterSpacing: "1.5px",
  },
  travelThumbnailsBlogTitle: {
    fontSize: {
      xs: "24px",
      xl: "32px",
    },
    margin: "12px 0px 6px",
    fontWeight: 500,
    whiteSpace: "nowrap",
    textOverflow: "ellipsis",
    overflow: "hidden",
  },
  travelThumbnailsLocationText: {
    fontSize: {
      xs: "14px",
      xl: "18px",
    },
    opacity: 0.8,
  },

  // blog group - stories & itineraries
  travelThumbnailsBlogGroupTitle: {
    textTransform: "uppercase",
    margin: 0,
    width: "100%",
    padding: "0rem 20px",
    boxSizing: "border-box",
    fontSize: "20px",
  },
}
