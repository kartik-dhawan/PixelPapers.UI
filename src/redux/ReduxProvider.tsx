"use client"

/**
 *
 * This component acts as a parent for client side components that require redux store
 * Wrap the component with this 'Redux Provider' to use state management on client side
 *
 * the updates made on the client side in the store will not be reflected in the server side components
 *
 * SERVER SIDE:
 * - Dispatching: store.dispatch(functionName(payload))
 * - Getting data from store: store.getState().sliaceName.stateName
 *
 * CLIENT SIDE:
 * - Dispatching: dispatch(functionName(payload)) [useDispatch()]
 * - Getting data from store: useSelector((state)=>state.slice.statename)
 *
 * Ref: https://www.youtube.com/watch?v=dRLjoT4r-jc
 */

import { Provider } from "react-redux"
import { store } from "./store"
import { ChildrenProps } from "@/utils/interfaces"

const ReduxProvider = ({ children }: ChildrenProps) => {
  return <Provider store={store}>{children}</Provider>
}

export default ReduxProvider
