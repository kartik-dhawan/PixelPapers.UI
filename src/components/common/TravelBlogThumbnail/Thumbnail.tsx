import { Box, ImageListItem, ImageListItemBar, Skeleton } from "@mui/material"
import { styles } from "./styles"
import Link from "next/link"
import Image from "next/image"
import { useCallback, useState } from "react"
import { dmSans } from "@/utils/fonts"

interface ThumbnailProps {
  item: any
  index: number
  largeCardIndex: 0 | 1 | 2
}

export const ThumbnailHeadings = ({ item }: any) => {
  return (
    <>
      <Box sx={styles.travelThumbnailsBlogType}>{item?.blogType ?? ""}</Box>
      <Box component="h3" sx={styles.travelThumbnailsBlogTitle}>
        {item?.blogMetaTitle ?? ""}
      </Box>
      <Box sx={styles.travelThumbnailsLocationText}>
        {item?.blogTravelLocation ?? ""}
      </Box>
    </>
  )
}

const Thumbnail = ({ item, index, largeCardIndex }: ThumbnailProps) => {
  const tid = "travelThumbnail"

  // sets the loading state of the images
  const [hasPlaceholder, setHasPlaceholder] = useState(true)

  const imageLoadingHandler = useCallback(() => {
    setHasPlaceholder(false)
  }, [])

  return (
    <ImageListItem
      className={tid + "CardListItem"}
      data-testid={tid + "CardListItem" + index}
      sx={{
        flexGrow: index === largeCardIndex ? 2 : 1,
        ...styles.travelThumbnailsCardListItem,
      }}
    >
      <Link
        href={`/travel/${item.blogUrlSlug}`}
        target="_blank"
        rel="noreferrer"
        aria-label={`${
          item?.blogMetaTitle ?? "Travel & More"
        }: Click to view the blog`}
      >
        <Box
          className={tid + "ImageWrapper"}
          data-testid={tid + "ImageWrapper" + index}
          sx={styles.travelThumbnailsImageWrapper}
        >
          {hasPlaceholder && (
            <Skeleton variant="rectangular" height="100%" width="100%" />
          )}
          <Image
            onLoad={imageLoadingHandler}
            src={`https:${item.blogMetaImage.fields.file.url}`}
            alt={item.blogMetaImage.fields.description}
            fill
            sizes="(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw)"
          />
        </Box>
        {!hasPlaceholder && (
          <ImageListItemBar
            className={tid + "TextWrapper " + dmSans.className}
            data-testid={tid + "TextWrapper" + index}
            sx={styles.travelThumbnailsCardTextWrapper}
            subtitle={<ThumbnailHeadings item={item} />}
          >
            <Box>{item?.blogType ?? ""}</Box>
          </ImageListItemBar>
        )}
      </Link>
    </ImageListItem>
  )
}

export default Thumbnail
