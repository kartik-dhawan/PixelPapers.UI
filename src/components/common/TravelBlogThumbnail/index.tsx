"use client"

import { RootType } from "@/redux/store"
import { useSelector } from "react-redux"
import { Box, ImageList } from "@mui/material"
import { ThumbnailItem } from "@/utils/interfaces"
import { styles } from "./styles"
import Thumbnail from "./Thumbnail"

interface TravelThumbnailProps {
  type: "stories" | "itineraries"
  largeCardIndex: 0 | 1 | 2
}

const TravelBlogThumbnail = ({
  type,
  largeCardIndex,
}: TravelThumbnailProps) => {
  const tid = "travelThumbnail"
  const { content } = useSelector((state: RootType) => state.contentSlice)

  return (
    <Box
      className={tid + "Wrapper"}
      id={tid + "Wrapper"}
      sx={styles.travelThumbnailsSectionWrapper}
    >
      <ImageList
        className={tid + "ImageList"}
        data-testid={tid + "ImageList"}
        sx={styles.travelThumbnailsImageList}
        cols={3}
        gap={12}
      >
        {content[
          type === "stories" ? "travelStoryBlogs" : "travelItineraryBlogs"
        ]
          ?.slice(0, 3)
          .map((item: ThumbnailItem, index: number) => {
            return (
              <Thumbnail
                key={item.sys.id}
                item={item.fields}
                index={index}
                largeCardIndex={largeCardIndex}
              />
            )
          })}
      </ImageList>
    </Box>
  )
}

export default TravelBlogThumbnail
