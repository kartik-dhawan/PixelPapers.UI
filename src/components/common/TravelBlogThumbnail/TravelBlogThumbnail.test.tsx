import ReduxProvider from "@/redux/ReduxProvider"
import "@testing-library/jest-dom"
import renderer from "react-test-renderer"
import TravelBlogThumbnail from "."
import Thumbnail, { ThumbnailHeadings } from "./Thumbnail"
import { TRAVEL_PAGE_THUMBNAIL_CARD_DATA } from "@/utils/mockData"

describe("Travel blog thumbnail cards component", () => {
  it("-- should match the snapshot", () => {
    const tree = renderer.create(
      <ReduxProvider>
        <TravelBlogThumbnail type="stories" largeCardIndex={2} />
      </ReduxProvider>,
    )

    expect(tree).toMatchSnapshot()
  })
})

describe("The thumbnail card component", () => {
  it("-- should match the snapshot", () => {
    const tree = renderer.create(
      <ReduxProvider>
        <Thumbnail
          item={TRAVEL_PAGE_THUMBNAIL_CARD_DATA}
          index={0}
          largeCardIndex={2}
        />
      </ReduxProvider>,
    )

    expect(tree).toMatchSnapshot()
  })
})

describe("The thumbnail text data component", () => {
  it("-- should match the snapshot", () => {
    const tree = renderer.create(
      <ReduxProvider>
        <ThumbnailHeadings item={TRAVEL_PAGE_THUMBNAIL_CARD_DATA} />
      </ReduxProvider>,
    )

    expect(tree).toMatchSnapshot()
  })
})
