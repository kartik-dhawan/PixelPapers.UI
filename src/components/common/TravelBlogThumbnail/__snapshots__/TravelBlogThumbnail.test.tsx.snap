// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The thumbnail card component -- should match the snapshot 1`] = `
<li
  className="MuiImageListItem-root travelThumbnailCardListItem css-96pj01-MuiImageListItem-root"
  data-testid="travelThumbnailCardListItem0"
  style={
    {
      "breakInside": undefined,
      "gridColumnEnd": "span 1",
      "gridRowEnd": "span 1",
      "height": "auto",
      "marginBottom": undefined,
    }
  }
>
  <a
    aria-label="Click to view the blog: Jibhi - Jalori"
    href="#"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
    rel="noreferrer"
    target="_blank"
  >
    <div
      className="travelThumbnailImageWrapper MuiBox-root css-1kzsnqq"
      data-testid="travelThumbnailImageWrapper0"
    >
      <span
        className="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-153n0d3-MuiSkeleton-root"
        style={
          {
            "height": "100%",
            "width": "100%",
          }
        }
      />
      <img
        alt="A picture of sunset from Raghupur fort close to jalori pass"
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        sizes="(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw)"
        src="/_next/image?url=https%3A%2F%2Fimages.ctfassets.net%2F1v98waf66sfj%2F6TVtXnjJIZ812Bxm6VBqz5%2F5b89ce3dced278570a2a663453d69a65%2FIMG_3749.webp&w=3840&q=75"
        srcSet="/_next/image?url=https%3A%2F%2Fimages.ctfassets.net%2F1v98waf66sfj%2F6TVtXnjJIZ812Bxm6VBqz5%2F5b89ce3dced278570a2a663453d69a65%2FIMG_3749.webp&w=256&q=75 256w, /_next/image?url=https%3A%2F%2Fimages.ctfassets.net%2F1v98waf66sfj%2F6TVtXnjJIZ812Bxm6VBqz5%2F5b89ce3dced278570a2a663453d69a65%2FIMG_3749.webp&w=384&q=75 384w, /_next/image?url=https%3A%2F%2Fimages.ctfassets.net%2F1v98waf66sfj%2F6TVtXnjJIZ812Bxm6VBqz5%2F5b89ce3dced278570a2a663453d69a65%2FIMG_3749.webp&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fimages.ctfassets.net%2F1v98waf66sfj%2F6TVtXnjJIZ812Bxm6VBqz5%2F5b89ce3dced278570a2a663453d69a65%2FIMG_3749.webp&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fimages.ctfassets.net%2F1v98waf66sfj%2F6TVtXnjJIZ812Bxm6VBqz5%2F5b89ce3dced278570a2a663453d69a65%2FIMG_3749.webp&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fimages.ctfassets.net%2F1v98waf66sfj%2F6TVtXnjJIZ812Bxm6VBqz5%2F5b89ce3dced278570a2a663453d69a65%2FIMG_3749.webp&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fimages.ctfassets.net%2F1v98waf66sfj%2F6TVtXnjJIZ812Bxm6VBqz5%2F5b89ce3dced278570a2a663453d69a65%2FIMG_3749.webp&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fimages.ctfassets.net%2F1v98waf66sfj%2F6TVtXnjJIZ812Bxm6VBqz5%2F5b89ce3dced278570a2a663453d69a65%2FIMG_3749.webp&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fimages.ctfassets.net%2F1v98waf66sfj%2F6TVtXnjJIZ812Bxm6VBqz5%2F5b89ce3dced278570a2a663453d69a65%2FIMG_3749.webp&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fimages.ctfassets.net%2F1v98waf66sfj%2F6TVtXnjJIZ812Bxm6VBqz5%2F5b89ce3dced278570a2a663453d69a65%2FIMG_3749.webp&w=3840&q=75 3840w"
        style={
          {
            "bottom": 0,
            "color": "transparent",
            "height": "100%",
            "left": 0,
            "objectFit": undefined,
            "objectPosition": undefined,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": "100%",
          }
        }
      />
    </div>
  </a>
</li>
`;

exports[`The thumbnail text data component -- should match the snapshot 1`] = `
[
  <div
    className="MuiBox-root css-nlooa6"
  >
    Travel Itinerary
  </div>,
  <h4
    className="MuiBox-root css-1liqna"
  >
    Jibhi - Jalori
  </h4>,
  <div
    className="MuiBox-root css-1g91hrd"
  >
    Himachal Pradesh
  </div>,
]
`;

exports[`Travel blog thumbnail cards component -- should match the snapshot 1`] = `
<div
  className="travelThumbnailWrapper MuiBox-root css-tb3nyz"
  id="travelThumbnailWrapper"
>
  <ul
    className="MuiImageList-root MuiImageList-standard travelThumbnailImageList css-gsuz1i-MuiImageList-root"
    data-testid="travelThumbnailImageList"
    style={
      {
        "gap": 12,
        "gridTemplateColumns": "repeat(3, 1fr)",
      }
    }
  >
    <p />
  </ul>
</div>
`;
